lib/ansible/plugins/callback/grafana_annotations.py

```diff
@@ -0,0 +1,203 @@
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+DOCUMENTATION = '''
+    callback: grafana_annotations
+    type: notification
+    short_description: Sends events to Grafana HTTP API
+    description:
+      - This callback will report start, failed and stats events to Grafana as annotations (https://grafana.com)
+    version_added: "2.5"
+    requirements:
+      - whitelisting in configuration
+    options:
+      server:
+        description: Address of the Logstash server
+        env:
+          - name: GRAFANA_SERVER
+        default: localhost
+      port:
+        description: Port on which Grafana is listening
+        env:
+            - name: GRAFANA_PORT
+        default: 3000
+      secure:
+        description: Boolean specifying if HTTPS should be used. (0=HTTP, 1=HTTPS)
+        env:
+          - name: GRAFANA_SECURE
+        default: 0
+      token:
+        description: Grafana API Token, allowing to authenticate when posting on the HTTP API.
+                     If not provided, the callback will be disabled.
+        env:
+          - name: GRAFANA_API_TOKEN
+'''
+
+import os
+import json
+import socket
+import getpass
+from datetime import datetime
+
+try:
+    import httplib
```

Comment:
you can use from `ansible.module_utils.urls import fetch_url` instead of import additional library

Reply:
The module_utils are for modules, here I'm submitting a callback.
The fetch_url function requires the module as first argument, I could try cheating by adding the params dict in my callback class, but the code also expects a fail_json method the is used to report the module failure.
That would be ugly imo.